File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        s = set(nums)
        count = 0
        n = len(nums)

        for n in nums:
            if n - 1 not in s:
                count = 1
                while n + count in s:
                    count += 1
                longest = max(longest, count)
            
        # i = 0
        # while i < n:
        #     count = 1
        #     while nums[i] + count in s:
        #         count += 1
    
        #     longest = max(longest, count)
        #     i += 1
        #     if longest > n - i + 1:
        #         break
        return longest

File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.rs
impl Solution {
    pub fn top_k_frequent(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let k = k as usize;
        let mut count = HashMap::new();
        for &num in &nums {
            *count.entry(num).or_insert(0i32) += 1;
        }

        let mut heap = BinaryHeap::new();
        for (&num, &freq) in &count {
            heap.push(Reverse((freq, num)));
            if heap.len() > k {
                heap.pop();
            }
        }

        heap.into_iter().map(|Reverse((_,num))| num).collect()
    }
}
